### PR TITLE
Improve support for triggers.

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

**Review by floitsch — APPROVED** [review]
LGTM

File: src/artemis.toit [hunk 1]
```diff
@@ -131,8 +151,34 @@ class TriggerTouch extends Trigger:
...
+  remaining-ms/int
+  constructor .remaining-ms:
+    super Trigger.KIND_RESTART
+
+  encode -> int:
+    return Trigger.encode-delayed remaining-ms
+
+  stringify -> string:
+    if remaining-ms == 0: return "Trigger - restart"
+    return "Trigger - restart in $(remaining-ms)ms"
```
Comment (kasperl) [inline src/artemis.toit:181]: We could have used a duration for this printing.

File: src/artemis.toit [hunk 2]
```diff
@@ -95,20 +95,36 @@ class Trigger:
...
+    return interval.in-ms << 5 | KIND_INTERVAL
+
+  static encode-delayed wakeup-time/int -> int:
+    return wakeup-time << 5 | KIND_RESTART
 
   static decode value/int -> Trigger?:
     if value == -1: return null
-    kind := value & 0xff
-    if kind == KIND_PIN: return TriggerPin (value >> 8)
+    kind := value & 0x1F
```
Comment (kasperl) [inline src/artemis.toit:115]: 0x1f perhaps?

Comment (floitsch) [reply src/artemis.toit:115]: https://github.com/toitware/artemis/pull/846

File: src/artemis.toit (hunk 1 above)
Comment (floitsch) [reply src/artemis.toit:181]: https://github.com/toitware/artemis/pull/846

**Pull request merged**